### PR TITLE
Firing analytics events when share buttons clicked

### PR DESCRIPTION
**Pull request by clintonb**
A tracking event is now sent to Segment when a user clicks any of the share buttons. This will allow us to track usage of these buttons.

ECOM-6578

### CODE REVIEW

**Comment by clintonb** [comment]
@edx/ecommerce please review

File: credentials/static/js/analytics.js [hunk 1]
```diff
@@ -11,4 +12,23 @@
     }
 
     analytics.page();
+
+    // Setup click handlers
+    for (let clickElement of document.querySelectorAll('[data-track-type="click"]')) {
```
Comment (schenedx) [inline credentials/static/js/analytics.js]: We have typescript translation on credential service?

Comment (clintonb) [inline credentials/static/js/analytics.js]: Nope. We have ES2017.

**Comment by schenedx** [comment]
No unit tests for those javascript functions?

File: credentials/static/js/analytics.js (hunk 1 above)
Comment (rlucioni) [inline credentials/static/js/analytics.js]: IE 11 does not support ES2017 features (http://kangax.github.io/compat-table/es2016plus/), but it doesn't look like you're using anything here that's new to ES2017. The `let` statement [is supported](https://docs.microsoft.com/en-us/scripting/javascript/reference/let-statement-javascript).

File: credentials/templates/_actions.html [hunk 2]
```diff
@@ -5,24 +5,37 @@
...
         </a>
       </button>
     {% endif %}
 
     {% if enable_linkedin_sharing %}
-      <button title="{% trans 'Add to LinkedIn profile' %}" class="action btn icon-only">
+      <button title="{% trans 'Add to LinkedIn profile' %}" class="action btn icon-only" data-track-type="click"
+              data-track-event="edx.bi.credentials.linkedin_share.attempted"
+              data-track-event-property-category="certificate"
+              data-track-event-property-credential-uuid="{{ user_credential.uuid }}"
```
Comment (rlucioni) [inline credentials/templates/_actions.html]: @clintonb have you been able to confirm that these properties are actually visible to those looking at GA? Historically, attributes beyond label, category, and action haven't been visible (meaning all this info was basically lost).

File: credentials/static/js/analytics.js (hunk 1 above)
Comment (clintonb) [inline credentials/static/js/analytics.js]: We use [Babel](https://babeljs.io/) to transpile ES2017 to browser-compatible JS.

**Comment by clintonb** [comment]
I have pretty much spent the entire day working on JS tests. At this point, they are not running on Travis. Having sunk a day of time on this, I can honestly say it's not worth continuing with that work given the limited usage of JavaScript in this repository. If someone wants to pickup where I left off, he/she can take https://github.com/edx/credentials/tree/js-tests. However, I have no plans to continue attempting to add tests to this PR. The value is extremely low.

**Comment by clintonb** [comment]
@edx/ecommerce please review
@AlasdairSwan please review the ESLint commit.

**Pull request merged**